src/sicks3000.cc

```diff
@@ -123,8 +123,13 @@ int SickS3000::Close()
     // Read controller messages.
     if (serial->ReadPort(cReadBuffer, &read_bytes, 1999) < 0)
```

Comment:
FWIW, the 500kbit/s was actually a misunderstanding on my part. It _does_ work, and the problem was to do with the self_test method making random extra connections to the serial port and freaking it out.

But by the time I had this all figured out, I'd already written a `select()` blocker for the old serial lib and didn't want to risk even more disruptive changes— as above, my preference would be to just rewrite this some lazy afternoon in the new year.